.github/workflows/build-assets-compilation.yml

```
@@ -97,17 +97,19 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-node-
 
-      - name: Extract package hash into env [DEV]
+      - name: Set env variables [DEV]
         if: ${{ !contains(github.ref, 'refs/tags/') }}
-        run: echo "ASSETS_HASH=$(composer assets-hash)" >> $GITHUB_ENV
-
-      - name: Extract tag name into env [PROD]
-        if: ${{ contains(github.ref, 'refs/tags/') }}
-        run: echo "TAG_NAME=$(echo ${GITHUB_REF#refs/*/})" >> $GITHUB_ENV
+        run: |
+          echo "ASSETS_HASH=$(composer assets-hash)" >> $GITHUB_ENV
+          echo "BUILD_ENV=dev" >> $GITHUB_ENV
 
-      - name: Set ENCORE_ENV to production [PROD]
+      - name: Set env variables [PROD]
         if: ${{ contains(github.ref, 'refs/tags/') }}
-        run: echo "ENCORE_ENV=production" >> $GITHUB_ENV
+        run: |
+          echo "TAG_NAME=$(echo ${GITHUB_REF#refs/*/})" >> $GITHUB_ENV
+          echo "BUILD_ENV=production" >> $GITHUB_ENV
+          echo "::notice::The ENCORE_ENV variable is deprecated and will be replaced in future by BUILD_ENV"
```

GitHub uses level "warning" for deprecations. Let's align with that.
```suggestion
          echo "::warning::The `ENCORE_ENV` variable is deprecated and will be removed soon. Please upgrade to using `BUILD_ENV`."
```